recipes/pcapplusplus/cmake/conanfile.py

```diff
@@ -0,0 +1,60 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake
+from conan.tools.files import copy, get, rmdir
+from conan.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.57.0"
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    package_type = "static-library"
+    license = "Unlicense"
+    description = "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets"
+    topics = ("pcap", "network", "security", "packet")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seladb/PcapPlusPlus"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {
+        "immediate_mode": [True, False],
+    }
+    default_options = {
+        "immediate_mode": False,
+    }
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("npcap/1.70")
+        else:
+            self.requires("libpcap/1.10.1")
+
+    def validate(self):
+        if self.settings.os not in ("FreeBSD", "Linux", "Macos", "Windows"):
+            raise ConanInvalidConfiguration(f"{self.settings.os} is not supported")
+
+    def source(self):
+        get(
+            self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True
+        )
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PCAPPP_BUILD_TESTS"] = False
+        tc.variables["PCAPPP_BUILD_EXAMPLES"] = False
+        tc.generate()
```

Comment:
No CMakeDeps? How dependencies are discovered?

Reply:
PcapPlusPlus doesn't use `find_package()` because it doesn't have dependencies that must be compiled with the library. Hence `CMakeDeps` is not needed

Reply:
@seladb but what about the libpcap listed as requirement?

Reply:
@uilianries `libpcap`/`WinPcap`/`Npcap` aren't compliled as part of PcapPlusPlus build system. It's assumed the compliled binaries already exist on the system:
- On macOS it's easy because libpcap comes as part of the OS
- On Linux a user can usually download and install libpcap using `apt` or `yum`
- On Windows the user should download [WinPcap developer's pack](https://www.winpcap.org/devel.htm) OR [Npcap SDK](https://nmap.org/npcap/guide/npcap-devguide.html)

You can read more about it here: https://pcapplusplus.github.io/docs/install